tests/test_backends.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+"""
+Tests for the `openedx-cas` backends module.
+"""
+import urllib.parse
+from unittest.mock import Mock, patch
+
+from ddt import data, ddt, unpack
+from django.conf import settings
+from django.test import TestCase, override_settings
+from social_core.exceptions import AuthFailed, AuthMissingParameter
+
+from openedx_cas.backends import CASAuth
+
+
+@ddt
+class TestOpenEdxCasAuthBackend(TestCase):
+    """
+    Test class to verify standard behavior of utility methods that belong to CASAuth.
+    """
+
+    def setUp(self) -> None:
+        super().setUp()
+        strategy = Mock()
+        self.cas_backend = CASAuth(strategy=strategy)
+
+    def test_cas_service_url_defined_in_settings(self):
+        """
+        This method is used to verify the CAS_SERVICE_URL variable in settings is defined
+        Expected behavior:
+        CASAuth.auth_url includes a query param called 'service' which value is a parsed string with the CAS_SERVICE_URL
+        """
+        auth_url = self.cas_backend.auth_url()
+        self.assertIn(urllib.parse.quote(settings.CAS_SERVICE_URL), auth_url)
+
+    @override_settings(
+        CAS_SERVER_URL=None
+    )
+    def test_cas_server_url_undefined_in_settings(self):
+        """
+        This method is used to verify the CAS_SERVER_URL variable in settings is defined
+        Expected behavior:
+        raise AuthMissingParameter
+        """
+        self.assertRaises(AuthMissingParameter, self.cas_backend.auth_url)
+
+    @override_settings(
+        CAS_SERVICE_URL=None
+    )
+    def test_cas_service_url_undefined_in_settings(self):
+        """
+        This method is used to verify the CAS_SERVICE_URL variable in settings is defined
+        Expected behavior:
+        raise AuthMissingParameter
+        """
+        self.assertRaises(AuthMissingParameter, self.cas_backend.auth_url)
+
+    @patch('openedx_cas.backends.CASAuth.cas_validation')
+    def test_auth_complete_ticket_validated(self, cas_validation):
+        """
+        This method is used to verify that the flow is valid when the ticket is valid
+        Expected behavior:
+        Response returned by auth_complete should include the correspoding values when the ticket is valid
+        """
+        cas_validation.return_value = {
+            'username': 'edxplatform'
+        }
+        service_response = {"ticket": "213123123-123123123"}
+        request = Mock(GET=service_response)
+        self.cas_backend.auth_complete(request=request)
+
+        self.cas_backend.strategy.authenticate.assert_called_with(request=request, response={'username': 'edxplatform'},
+                                                                  backend=self.cas_backend)
```

Comment:
improve arrangement:
```suggestion
self.cas_backend.strategy.authenticate.assert_called_with(
    request=request,
    response={'username': 'edxplatform'},
    backend=self.cas_backend,
)
```